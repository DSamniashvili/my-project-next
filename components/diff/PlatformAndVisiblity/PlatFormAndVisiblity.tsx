import {FC} from 'react';
import {
   SPlatformAndVisbilityContainer,
   SPlatformAndVisbilityDivider,
   SPlatformAndVisibilityContentContainer,
   SPlatformAndVisibilityWrapper,
} from './SPlatformAndVisiblity.styled';
import {PlatformAndVisbilityContent} from './PlatformAndVisbilityContent/PlatformAndVisbilityContent';
import {PlatFormAndVisiblityProps} from './PlatFormAndVisiblity.types';
import {useDeviceDetect} from '@hooks/useDeviceDetect';

const PlatFormAndVisiblity: FC<PlatFormAndVisiblityProps> = () => {
   const {isMobile} = useDeviceDetect();
   const platformAndVisibilityObj = [
      {
         id: 1,
         title: 'platform',
         description:
            'See the limitless opportunities to organize and manage the supply chain with less clicks and more logistics.',
         buttonText: 'get your demo',
         url: '/',
      },
      {
         id: 2,
         title: 'visibility',
         description:
            'Sit back, relax, and enjoy the visualization of your cargo in transit through OMOFOX platform.',
         buttonText: 'Join now',
         url: '/',
      },
   ];

   return (
      <SPlatformAndVisbilityContainer>
         <SPlatformAndVisibilityWrapper>
            <SPlatformAndVisibilityContentContainer>
               {platformAndVisibilityObj.map(
                  ({id, title, description, buttonText}, index) => {
                     return (
                        <div key={id} className={'flex'}>
                           <div>
                              <PlatformAndVisbilityContent
                                 title={title}
                                 description={description}
                                 buttonText={buttonText}
                              />
                           </div>
                           {index === 0 && !isMobile && (
                              <SPlatformAndVisbilityDivider />
                           )}
                        </div>
                     );
                  },
               )}
            </SPlatformAndVisibilityContentContainer>
         </SPlatformAndVisibilityWrapper>
      </SPlatformAndVisbilityContainer>
   );
};

export default PlatFormAndVisiblity;
