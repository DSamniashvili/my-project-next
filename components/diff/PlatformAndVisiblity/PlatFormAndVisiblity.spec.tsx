import {render, screen} from '@testing-library/react';
import PlatFormAndVisiblity from './PlatFormAndVisiblity';

describe('[Component] PlatFormAndVisiblity', () => {
   it('should render platform transltions correctly', () => {
      const {container} = render(<PlatFormAndVisiblity />);
      expect(container).toMatchSnapshot();
   });
});
