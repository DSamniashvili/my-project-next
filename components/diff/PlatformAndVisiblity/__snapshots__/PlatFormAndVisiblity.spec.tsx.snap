// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] PlatFormAndVisiblity should render platform transltions correctly 1`] = `
<div>
  <section
    class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityContainer-sc-vqbjye-0 jhlzeT"
  >
    <div
      class="SPlatformAndVisiblitystyled__SPlatformAndVisibilityWrapper-sc-vqbjye-1 iWdPqR container"
    >
      <div
        class="SPlatformAndVisiblitystyled__SPlatformAndVisibilityContentContainer-sc-vqbjye-2 fdDKNy"
      >
        <div
          class="flex"
        >
          <div>
            <div
              class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityContent-sc-vqbjye-5 iPcflY"
            >
              <h2
                class="SPlatformAndVisiblitystyled__SPlatformAndVisiblityHeader-sc-vqbjye-6 hWhFhC"
              >
                platform
              </h2>
              <p
                class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityParagraph-sc-vqbjye-4 gAcEjJ"
              >
                See the limitless opportunities to organize and manage the supply chain with less clicks and more logistics.
              </p>
              <button
                class="AppButtonstyled__SAppButton-sc-i11sy7-0 TuVHs"
                role="button"
              >
                get your demo
              </button>
            </div>
          </div>
          <div
            class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityDivider-sc-vqbjye-3 juQcHF"
          />
        </div>
        <div
          class="flex"
        >
          <div>
            <div
              class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityContent-sc-vqbjye-5 iPcflY"
            >
              <h2
                class="SPlatformAndVisiblitystyled__SPlatformAndVisiblityHeader-sc-vqbjye-6 hWhFhC"
              >
                visibility
              </h2>
              <p
                class="SPlatformAndVisiblitystyled__SPlatformAndVisbilityParagraph-sc-vqbjye-4 gAcEjJ"
              >
                Sit back, relax, and enjoy the visualization of your cargo in transit through OMOFOX platform.
              </p>
              <button
                class="AppButtonstyled__SAppButton-sc-i11sy7-0 TuVHs"
                role="button"
              >
                Join now
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
